File: Dictionaries and Sets/771. Jewels and Stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        j = set(jewels)
        c = 0
        for s in stones:
            if s in j:
                c += 1
        return c

# Time: O(n + m)
# Space: O(n)

File: Dictionaries and Sets/771. Jewels and Stones/771. Jewels and Stones.js
/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function (jewels, stones) {
  const j = new Set(jewels);
  let c = 0;
  for (let s of stones) {
    if (j.has(s)) {
      c++;
    }
  }
  return c;
};

// Time: O(n + m)
// Space: O(m)
